openedx/core/djangoapps/theming/core.py

```diff
@@ -2,66 +2,80 @@
 Core logic for Comprehensive Theming.
 """
 
+from os.path import basename
+
 from django.conf import settings
 
+from path import Path
+
 import edxmako
 
 
-def comprehensive_theme_changes(theme_dir):
+def get_configuration(path_theme):
     """
-    Calculate the set of changes needed to enable a comprehensive theme.
+    Calculate the configuration needed to enable a comprehensive theme.
 
     Arguments:
-        theme_dir (path.path): the full path to the theming directory to use.
+        path_theme (path.Path): the full path to the theming directory to use.
 
     Returns:
-        A dict indicating the changes to make:
+        A dict indicating the configuration to apply:
 
             * 'settings': a dictionary of settings names and their new values.
 
             * 'mako_paths': a list of directories to prepend to the edxmako
                 template lookup path.
-
     """
-
-    changes = {
+    configuration = {
         'settings': {},
         'mako_paths': [],
     }
-
-    if settings.PROJECT_ROOT.endswith("cms"):
```

Comment:
This is brittle; a path like `themes/MY_THEME/lms/override_from_cms/` will be incorrectly categorized.

Further, maintaining a mapping between "studio" and "cms" throughout the codebase is more likely to break than just using path names consistently.